### PR TITLE
Fix calculation

### PR DESCRIPTION
**Pull request by myluki2000**
## Description of changes
Previously, the ContentServiceClient returned content data in a format which left types mangled and data unset. This caused skilllevel calculation to fail. This PR fixes the ContentServiceClient such that it returnes data in the proper format.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [x] manual, exploratory test
No automated tests for ContentServiceClient as we plan to move this code to a central location (currently there exist multiple ContentServiceClients in different repos)
    
## Checklist before requesting a review
- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/ContentServiceClient.java [hunk 1]
```diff
@@ -66,69 +64,107 @@ public List<Content> getContentsWithUserProgressData(UUID userId,
...
-                    .flatMap(Arrays::stream)
-                    .map(ContentWithUserProgressData::toContent)
-                    .toList();
+            return result;
         } catch (Exception e) {
             throw new ContentServiceConnectionException("Error while fetching contents from content service", e);
         }
     }
 
+    public static class GenericAssessmentResponse {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/ContentServiceClient.java]: Why not make it a record? Or at least use `@Getter` and `@Setter`?

File: src/test/java/de/unistuttgart/iste/gits/skilllevel_service/SkillLevelServiceTest.java [hunk 2]
```diff
@@ -277,7 +278,14 @@ void testRepetitionNotEnoughTimePassed() {
                         .build()
         );
 
-        when(contentServiceClient.getContentsWithUserProgressData(any(), any())).thenReturn(contents);
+        when(contentServiceClient.getAssessmentsOfChapter(any(), any())).thenReturn(
+                contents.stream()
+                        .map(x -> new ContentServiceClient.GenericAssessmentResponse(
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/skilllevel_service/SkillLevelServiceTest.java]: if you add a constructor to "GenericAssessmentResponse" that accepts a "Assessment" you don't need this construct.

File: src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/ContentServiceClient.java [hunk 3]
```diff
@@ -66,69 +64,107 @@ public List<Content> getContentsWithUserProgressData(UUID userId,
...
-                                learningInterval
-                                nextLearnDate
-                                lastLearnDate
-                                log {
-                                    timestamp
-                                    success
-                                    correctness
-                                    hintsUsed
-                                    timeToComplete
+                            ...on Assessment {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/ContentServiceClient.java]: What happens with media content? then it will not return anything or how does it work?

File: src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/calculation/SkillLevelCalculator.java [hunk 4]
```diff
@@ -34,34 +32,32 @@ public class SkillLevelCalculator {
...
         allSkillLevelsEntity.setApply(new SkillLevelEntity(0));
 
-        return calculate(allSkillLevelsEntity, contents);
+        return calculate(allSkillLevelsEntity, assessments);
     }
 
-    private AllSkillLevelsEntity calculate(AllSkillLevelsEntity allSkillLevelsEntity, List<Content> contents) {
-        if(contents.isEmpty()) {
+    private AllSkillLevelsEntity calculate(AllSkillLevelsEntity allSkillLevelsEntity,
+                                           List<ContentServiceClient.GenericAssessmentResponse> assessments) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/calculation/SkillLevelCalculator.java]: I don't really like that you use the rather internal mapping class here outside of the client class. I personally would make "GenericAssessmentResponse" implement the "Assessment" interface, then you can just use Assessment here

**Review by PaulBredl — CHANGES_REQUESTED** [review]
I have one question regarding how the client behaves with MediaContent and some #Nitpicks

File: src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/ContentServiceClient.java (hunk 1 above)
Comment (myluki2000) [reply src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/ContentServiceClient.java]: Originally I used inheritance which doesn't work with records I think. I changed it to use inheritance now so the class stays, but you are obviously correct that i can at least use the annotations

File: src/test/java/de/unistuttgart/iste/gits/skilllevel_service/SkillLevelServiceTest.java (hunk 2 above)
Comment (myluki2000) [reply src/test/java/de/unistuttgart/iste/gits/skilllevel_service/SkillLevelServiceTest.java]: also not necessary if the ContentResponse class implements Content which i did now

File: src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/ContentServiceClient.java (hunk 3 above)
Comment (myluki2000) [reply src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/ContentServiceClient.java]: my solution actually does not work. I thought I tested it with media contents in the chapter but I didn't. Pushed a new version.

File: src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/calculation/SkillLevelCalculator.java (hunk 4 above)
Comment (myluki2000) [reply src/main/java/de/unistuttgart/iste/gits/skilllevel_service/service/calculation/SkillLevelCalculator.java]: yes

**Review by PaulBredl — APPROVED** [review]
